shared/api/storage_pool_volume.go

```diff
@@ -120,6 +122,12 @@ type StorageVolume struct {
 	ContentType string `json:"content_type" yaml:"content_type"`
 }
 
+// URL returns the URL for the storage volume.
+func (storageVolume *StorageVolume) URL(apiVersion string, pool string, project string, target string) *URL {
+	volPath := fmt.Sprintf("%s/volumes/%s/%s", pool, storageVolume.Type, storageVolume.Name)
```

Comment:
This commit can be a separate PR.

Also, why are you building volPath in an unescaped way rather than passing the 3 params to `Path()` function below it?